2016/proposals/stolliker-ryan.md

```diff
@@ -0,0 +1,67 @@
+# Ryan Stolliker: Result Aggregation Server for Software Carpentry
+
+## Abstract
+
+During Software Carpentry's workshops which teach computing skills to the greater scientific community, learners use Python scripts to test that they have installed the programming environment and other dependencies correctly. Currently, this information is known only to the participants and cannot be used or tracked by the organizers. Thus, the goal of this project is to enable the collection of this installation data and to store it for future use and analysis.
+
+## Technical Details
+
+### Server
+
+All server management will be written in Python. The server itself will be implemented using [bottle](http://bottlepy.org/docs/dev/index.html), a simple web framework with functionality to parse forms and JSON requests and responses. SQLite will be used to create and manage the database which holds the results.
+
+### Updates to Test Scripts
+
+The existing installation testing scripts, already written in Python, will be updated to send data to the server. Because the point of the scripts is to determine whether Python is configured correctly, the changes should not rely on any third party libraries and be compatible with both Python 2 and 3 so that even if a system fails a test, it will still be able to send the results. System data will be collected using the process library, which is included in Python's standard library. Data will be sent using the http.client and urllib.parse libraries.
```

Comment:
I'd like to see how you envision communication between scripts and the server. It can be a draft of API or examples of actual calls to the server (and responses they'd get).